flowzone.yml

```diff
@@ -2997,6 +3065,21 @@ jobs:
             poetry add --group dev flake8@latest pydocstyle@latest pytest@latest
           fi
 
+      - name: Generate metadata
+        id: meta
+        run: |
+          package=$(grep '^name =' pyproject.toml | sed 's/name = "\(.*\)"/\1/')
+          version=$(grep '^version =' pyproject.toml | sed 's/version = "\(.*\)"/\1/')
+          branch_tag="$(echo "build-${GITHUB_HEAD_REF}" | sed 's/[^[:alnum:]]/-/g')"
+          sha_tag="${branch_tag}-${{ github.event.pull_request.head.sha }}"
+          version_tag="${version}-${branch_tag}-${{ github.event.pull_request.head.sha }}"
+
+          echo "package=${package}" >> "${GITHUB_OUTPUT}"
+          echo "version=${version}" >> "${GITHUB_OUTPUT}"
+          echo "branch_tag=${branch_tag}" >> "${GITHUB_OUTPUT}"
+          echo "sha_tag=${sha_tag}" >> "${GITHUB_OUTPUT}"
+          echo "version_tag=${version_tag}" >> "${GITHUB_OUTPUT}"
```

Comment:
Do you mean between languages?

It could probably be done. AFAIU would need to split `package` and `version` in a language-specific step and pass the `version` as an env to the shared one.

Reply:
I thought I saw another python job using the same function but I must have been confused. It's fine as it is.